tracer/src/Datadog.Trace/Ci/CIVisibility.cs

```diff
@@ -525,6 +538,49 @@ internal static void Reset()
 
         private static async Task ShutdownAsync()
         {
+            // Let's close any opened test, suite, modules and sessions before shutting down to avoid losing any data.
+            // But marking them as failed.
+
+            foreach (var test in Test.ActiveTests)
+            {
+                if (LifetimeManager.Instance.CurrentException is { } exception)
```

Comment:
I discuss changing this pattern later, but it probably makes sense to pull this outside each of the loops and just checking for null instead of repeated static access?

```suggestion
           var exception = LifetimeManager.Instance.CurrentException;
            foreach (var test in Test.ActiveTests)
            {
                if (exception is not null)
```

Reply:
done in [7609a88](https://github.com/DataDog/dd-trace-dotnet/pull/5549/commits/7609a88f288a9dd6ff6ab26f4a7675458ab21bab)